core/src/consensus.rs

```diff
@@ -651,6 +945,144 @@ impl Tower {
             None
         }
     }
+
+    pub fn get_filename(path: &Path, node_pubkey: &Pubkey) -> PathBuf {
+        path.join(format!("tower-{}", node_pubkey))
+            .with_extension("bin")
+    }
+
+    pub fn get_tmp_filename(path: &Path) -> PathBuf {
+        path.with_extension("bin.new")
+    }
+
+    pub fn save(&self, node_keypair: &Arc<Keypair>) -> Result<()> {
+        let mut measure = Measure::start("tower_save-ms");
+
+        if self.node_pubkey != node_keypair.pubkey() {
+            return Err(TowerError::WrongTower(format!(
+                "node_pubkey is {:?} but found tower for {:?}",
+                node_keypair.pubkey(),
+                self.node_pubkey
+            )));
+        }
+
+        let filename = &self.path;
+        let new_filename = &self.tmp_path;
+        {
+            // overwrite anything if exists
+            let mut file = File::create(&new_filename)?;
+            let saved_tower = SavedTower::new(self, node_keypair)?;
+            bincode::serialize_into(&mut file, &saved_tower)?;
+            // file.sync_all() hurts performance; pipeline sync-ing and submitting votes to the cluster!
```

Comment:
I'm aware of https://github.com/solana-labs/solana/pull/9902#discussion_r442439449, but I just noticed this introduces moderate latencies even with casual testing and local_cluster tests. So, I'm commenting out this for now.

I'll address the pipe-lining if this durability is really desired. Well, assuming most of validators are hosted at data-center, the power-outages and/or kernel-crashes should be very rare.

Reply:
yeah agreed. Out of curiosity, do you know about how long this `sync_all` was taking?

Reply:
@carllin Well, this depends on various factors including hardware, os block system config, filesystem, etc. On my machine, enabling this easily caused 1~2s spikes...